bench/micro/Bench/Database/LSMTree/Monoidal.hs

```diff
@@ -107,85 +106,87 @@ benchInsertsVsMupserts =
-    env (pure $ snd $ randomEntriesGrouped 800_000 250) $ \ess -> bgroup "lookups-inserts-vs-mupserts" [
-        env (pure $ V.map mkMonoidalInserts ess) $ \inss -> bench "lookups-inserts" $
-          withMonoidalTable inss $ \(_, _, _, _, t) ->
-            -- Insert the same keys again, but we sum the existing values in
-            -- the table with the values we are going to insert: first lookup
-            -- the existing values, sum those with the insert values, then
-            -- insert the updated values.
-            V.forM_ inss $ \ins -> do
-              lrs <- Monoidal.lookups t (V.map fst ins)
-              let ins' = V.zipWith f ins lrs
-              Monoidal.inserts t ins'
-      , env (pure $ V.map mkMonoidalMupserts ess) $ \mupss -> bench "mupserts" $
-          withMonoidalTable mupss $ \(_, _, _, _, t) ->
-            -- Insert the same keys again, but we sum the existing values in
-            -- the table with the values we are going to insert: submit
-            -- mupserts with the insert values.
-            V.forM_ mupss $ \mups -> Monoidal.mupserts t mups
-      ]
+    env (pure $ snd $ randomEntriesGrouped 800_000 250) $ \ess ->
+      env (pure $ V.map mkInserts ess) $ \inss ->
+        bgroup "lookups-inserts-vs-mupserts" [
+          bench "lookups-inserts" $
+            withMonoidalTable inss $ \(_, _, _, _, t) ->
+              -- Insert the same keys again, but we sum the existing values in
+              -- the table with the values we are going to insert: first lookup
+              -- the existing values, sum those with the insert values, then
+              -- insert the updated values.
+              V.forM_ ess $ \es -> do
+                lrs <- lookups t (V.map fst es)
+                let ins' = V.zipWith f es lrs
+                inserts t ins'
+        , bench "mupserts" $
+            withMonoidalTable inss $ \(_, _, _, _, t) ->
+              -- Insert the same keys again, but we sum the existing values in
+              -- the table with the values we are going to insert: submit
+              -- mupserts with the insert values.
+              V.forM_ ess $ \es -> mupserts t es
+        ]
   where
     f (k, v) = \case
-      Monoidal.NotFound -> (k, v)
-      Monoidal.Found v' -> (k, v `resolve` v')
+      NotFound -> (k, v, Nothing)
+      Found v' -> (k, v `resolve` v', Nothing)
 
     withMonoidalTable inss = perRunEnvWithCleanup
           -- Make a monoidal table and fill it up
           (do (tmpDir, hfs, hbio) <- mkFiles
-              (s, t) <- mkMonoidalTable hfs hbio benchConfig
-              V.mapM_ (Monoidal.inserts t) inss
+              (s, t) <- mkTable hfs hbio benchConfig
+              V.mapM_ (inserts t) inss
               pure (tmpDir, hfs, hbio, s, t)
           )
           (\(tmpDir, hfs, hbio, s, t) -> do
-              cleanupMonoidalTable (s, t)
+              cleanupTable (s, t)
               cleanupFiles (tmpDir, hfs, hbio)
           )
```

Comment:
We can call it `withTable`

Reply:
Fixed in a26bc18bd1ae692dd99e76d87f96e69607270052.